subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/DefaultJavaToolchainDownload.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain.internal;
+
+import org.gradle.jvm.toolchain.JavaToolchainDownload;
+
+import java.net.URI;
+
+public class DefaultJavaToolchainDownload extends JavaToolchainDownload {
+
+    public static DefaultJavaToolchainDownload fromUri(URI uri) {
```

Comment:
The same method exists in the parent. I would not add it here as well.

Reply:
My thinking was to make the constructor of `DefaultJavaToolchainDownload` private, hence I needed a static method there too. But indeed, it's not the best...